fili-core/src/test/groovy/com/yahoo/bard/webservice/logging/blocks/PrefaceSpec.groovy

```diff
@@ -42,6 +49,65 @@ class PrefaceSpec extends Specification {
         }
 
         expect: "Making a Preface with it sets the user to the user name"
-        new Preface(mockRequest).user == knownUserName
+        def preface = new Preface(mockRequest)
+        preface.getUser() == knownUserName
+        MAPPER.valueToTree(preface).toString() == """{"uri":"http://example.com","method":"","headers":{"Cookie":["Cookies not present in header"]},"user":"${knownUserName}"}"""
+    }
+
+    def "User set to username value at serialization time"() {
+        given: "A request with a userPrincipal"
+        def securityContext = new MockSecurityContext(null)
+        mockRequest.getSecurityContext() >> securityContext
+        def preface = new Preface(mockRequest)
+
+        expect: "Use default username value"
```

Comment:
It might be more clear what's going on here with a call to `securityContext.setUserPrincipal(null)`